src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,264 @@
+package duke.parser;
+
+import duke.exceptions.InvalidDeadlineException;
+import duke.exceptions.InvalidEventException;
+import duke.exceptions.InvalidKeyException;
+import duke.exceptions.InvalidRequestException;
+import duke.tasks.Event;
+import duke.tasks.Deadline;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.tasks.Todo;
+import duke.storage.Storage;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Interpret the command and do the instruction.
+ */
+public class Parser {
```

Comment:
Currently my understanding is that all important functionality (reading from standard input, parsing the command  and handling the logic) is in this `Parser` class. Perhaps consider whether this is optimal and more abstraction can be used?